### DISCUSSION TITLE
I need to setup ArgoCD DR fully automated in multi-region. Help me with the best approach as I am new to ArgoCD

### DISCUSSION THREAD

**Discussion by Harshith-Krishnappa96** [argoproj/argo-cd#15229] [Q&A]
Hi Team,

I have the ArgoCD installed in EKS Cluster. I have to set-up Disaster Recovery for the application in multi-region. I have tried exploring and have got this official document:
https://argo-cd.readthedocs.io/en/stable/operator-manual/disaster_recovery/ 

But the ask here is, I have to automate the complete DR process. Please help me with the workflow how best I can carry out this.

**Comment by FernandoMiguel** [comment]
Although I am a strong advocate of DR, for stuff like Argo that can, and should, be entirely codified, as should their applications, what is the benefit of DR?
We have destroyed Argo installs many times, only to spin a bran new one moments later that adopt the existing clusters and their running workloads with zero impact.

**Comment by Harshith-Krishnappa96** [comment]
Thanks @FernandoMiguel for your response. So rather than having a DR, we can have the backups at regular interval and store it. We can restore it using the backup when there is an issue in the primary ArgoCD is what you suggest?

**Reply by FernandoMiguel** [reply]
Quite the opposite.
If all your config is in code, what are you backing up and restoring?
If your Argo fails for any reason, just nuke it, treat it like cattle not as a pet.
Simply replace the install with a new one and you are off to the races minutes later.

In the past, when you restored a backup, how long did that take you, @Harshith-Krishnappa96?

**Comment by Harshith-Krishnappa96** [comment]
I tried taking a backup of Argo by export command and tried importing it in a new cluster by installing Argo. It took me 15 to 20 minutes for this procedure. Below are the commands I used:
# take complete backup of existing argocd
argocd admin export -n argocd > argocd-backup-$(date +"%Y-%m-%d_%H:%M").yml

argocd admin import - < argocd-backup-2023-07-19_16\:23.yml
kubectl apply -f argocd-backup-2023-07-19_16\:23.yml -n argocd

**Reply by FernandoMiguel** [reply]
Plus the time to install Argo, right?

Now imagine that just as you finish installing it, it's already managing your workloads....
Better?


@Harshith-Krishnappa96 how are you installing and managing Argo and the workloads?
Is it all in code?

**Reply by bhoppeadoy** [reply]
@FernandoMiguel can you explain this scenario in more detail? I'm looking at the similar scenario.

You say that after installing Argo, it just begins managing workloads....but how? It has no knowledge of anything after a fresh install

How would ArgoCD be able to start managing applications without some kind of restore? How does ArgoCD know what applications to deploy, if you haven't supplied ArgoCD with the application manifests?

Also, what about the projects, repositories (credential templates) that were created in ArgoCD. If you don't restore these from a backup, then how does the new ArgoCD install get these?

**Reply by bhoppeadoy** [reply]
If what you're saying is to handle all the configuration as code (i.e. manifests stored in git), and to not use the UI for any configuration changes, then I agree with what you're saying. You won't need a backup.

But, when "restoring" ArgoCD after a fresh install, it still requires applying the stored configuration before ArgoCD can go about managing anything.

**Reply by FernandoMiguel** [reply]
Correct. It's all code. Ui is read only.
Our bootstrap terraform module also configures Argo to read a git repo with the app sets.

**Comment by Harshith-Krishnappa96** [comment]
I am installing the Argo using Helm package  in the cluster and have integrated with GitLab which consists of the application code

**Comment by Harshith-Krishnappa96** [comment]
@FernandoMiguel : Thanks a lot for your suggestion. It helped me.
